### ISSUE TITLE
Breaking API change "thumbnail": "image"

### ISSUE THREAD

**Issue by QuantumBadger** [reddit/reddit#1692]
Reddit is now sending `"thumbnail": "image"` for some of the posts in the post listing. This breaks API clients which try to parse this as a URL.

I've released an update for RedReader which should fix this (on Google Play within the next few hours), but there are still users on old versions who will experience crashes every time they try to view a post list.

Since this is an unannounced, undocumented, breaking change to the API, it would be a good idea to revert it, and convey this information in a different way. The `post_hint` field already seems to be doing so.

**Comment by QuantumBadger** [comment]
I take it this was caused by the following:

https://www.reddit.com/r/changelog/comments/51u5gj/reddit_change_new_thumbnail_art_expando_art_and/

**Comment by madbook** [comment]
The API has returned either an image URL _or_ a string (indicating a default thumbnail image) for as long as I can remember.  I _could_ be wrong, but I don't think we even enumerate anywhere what those strings could be.  My suggestion would be to check if the string looks like a URL [here](https://github.com/QuantumBadger/RedReader/blob/master/src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedPost.java#L629-L638) (e.g. checking if it starts with "http" ought to be fine) and ignore all other cases.

**Comment by Pokechu22** [comment]
Yea, it's always had non-url values and you've parsed it like that before.  [Here's the current code that's being used to generate it on the open-source version](https://github.com/reddit/reddit/blob/8cc3f46ad9ba221a481069293a3f3f8c9800260d/r2/r2/models/link.py#L644-L678).  It's able to return:

* The actual URL
* An empty string
* The string `nsfw`
* The string `self`
* The string `default`

I'm guessing that the recent change makes it possible for it to also return `image`.

**Comment by QuantumBadger** [comment]
Yep, looks like you're right. I had to rush to fix this over breakfast this morning, so didn't realise the app was already looking for those other strings :)

While it's ultimately my fault for not making the app a bit more robust to this, I don't think it's a great idea to change the values an existing field can take on, particularly when the documentation is lacking. I managed to find this for the thumbnail field:

> full URL to the thumbnail for this link; "self" if this is a self post; "default" if a thumbnail is not available

https://github.com/reddit/reddit/wiki/JSON

So at the very least the docs need updating :)

Changing the behaviour of an existing field is poor API etiquette, and while it's a bit too late to suggest that the API be versioned to prevent things like this, it would still be good to maintain the previous behaviour of the field to avoid breaking old clients.
